src/src/Helpers/Rules_calculate.cpp

```diff
@@ -280,8 +280,13 @@ ESPEASY_RULES_FLOAT_TYPE RulesCalculate_t::apply_quinary_operator(char op,
   ESPEASY_RULES_FLOAT_TYPE ret{};
   const UnaryOperator qu_op = static_cast<UnaryOperator>(op);
 
-  if (UnaryOperator::Map == qu_op) {
-    return mapADCtoFloat(first, second, third, fourth, fifth);
+  if (UnaryOperator::Map == qu_op || UnaryOperator::MapC == qu_op) {
+    ret = mapADCtoFloat(first, second, third, fourth, fifth);
+
+    // Clamp the result if the operator is MapC
+    if (qu_op == UnaryOperator::MapC) {
+      ret = std::clamp(ret, std::min(fourth, fifth), std::max(fourth, fifth));
+    }
```

Comment:
I can't think of a counter-example to prove one is better than the other, but I would expect this `mapc` to clamp the input value to the given range of the input values.

Reply:
For me it was intuitively the output that needs to be clamped. In the end it does not make much of a difference...